src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java

```diff
@@ -137,39 +136,41 @@ public Payment getPlatformPayment(String name) {
      * @param experiment The experiment to publish
      * @return None if the platform does not exist
      */
+    //TODO improve doc: what is none? Completable with null (bad api) what does true/false mean? or does it return null?
     public CompletableFuture<Boolean> publishTask(String name, Experiment experiment) throws TaskOperationException {
-
-        TaskRecord record = new TaskRecord();
-        record.setExperiment(experiment.getId());
-        record.setStatus(TaskStatus.running);
-        record.setCrowdPlatform(name);
-
-        TaskRecord result = tasksOps.createTask(record);
-        if (result == null)
-            throw new TaskOperationException("Task could not be created");
+        ExperimentsPlatformRecord record = experimentsPlatformOps.getExperimentsPlatform(name, experiment.getId()).
+                orElseThrow(() -> new IllegalStateException("Platform is not activated for experiment " + experiment));
+
+        BiFunction<String, Throwable, Boolean> handlePublishResult = (s1, throwable) -> {
+            //if the creation was successful update the db
+            if (s1 != null && throwable == null && !s1.isEmpty()) {
+                experimentsPlatformOps.setPlatformStatus(record.getIdexperimentsPlatforms(),
+                        ExperimentsPlatformStatusPlatformStatus.running);
+                record.setPlatformData(s1);
+                if (!experimentsPlatformOps.updateExperimentsPlatform(record)) {
+                    throw new IllegalStateException("Updating record for published experimentsPlatform failed");
```

Comment:
fixed with 7f1d9e3ea633c98d4f592ac521ec4c3c980bd5c8, now it gets unpublished if the db-connection fails